controllers/sdkutil/spec_sdk_converters.go

```diff
@@ -584,3 +593,201 @@ func ConvertMapToKeyValuePairSlice(m map[string]string) []*commonv1.KeyValuePair
+
+	for _, resourceIDfromSpec := range resourceIDListfromSpec {
+		ResourceID := ConvertAutoscalingResourceToString(*resourceIDfromSpec)
+		output, _ = createPutScalingPolicyInputFromSpec(spec, ResourceID)
+		outputList = append(outputList, output)
+	}
+
+	return outputList, nil
+}
+
+// createPutScalingPolicyInputFromSpec request input from a Kubernetes spec.
+func createPutScalingPolicyInputFromSpec(spec hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJobSpec, resourceID *string) (applicationautoscaling.PutScalingPolicyInput, error) {
+	var output applicationautoscaling.PutScalingPolicyInput
+
+	marshalledPutScalingPolicyInputInput, err := json.Marshal(spec)
+	if err != nil {
+		return applicationautoscaling.PutScalingPolicyInput{}, err
+	}
+
+	if err = json.Unmarshal(marshalledPutScalingPolicyInputInput, &output); err != nil {
+		return applicationautoscaling.PutScalingPolicyInput{}, err
+	}
+
+	output.ResourceId = resourceID
+
+	return output, nil
+}
+
+// CreateDeregisterScalableTargetInput creates input
+func CreateDeregisterScalableTargetInput(spec hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJobSpec, resourceID string) applicationautoscaling.DeregisterScalableTargetInput {
+
+	if input, err := createDeregisterScalableTargetInput(spec, resourceID); err == nil {
+		return input
+	} else {
+		panic("Unable to create CreateDegisterScalableTargetInput " + err.Error())
+	}
+}
+
+// createDeregisterScalableTargetInput request input from a Kubernetes spec.
+func createDeregisterScalableTargetInput(spec hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJobSpec, resourceID string) (applicationautoscaling.DeregisterScalableTargetInput, error) {
+	var output applicationautoscaling.DeregisterScalableTargetInput
+
+	marshalledScalableDimension, err := json.Marshal(spec.ScalableDimension)
+	if err = json.Unmarshal(marshalledScalableDimension, &output.ScalableDimension); err != nil {
+		return applicationautoscaling.DeregisterScalableTargetInput{}, err
+	}
+
+	output.ResourceId = &resourceID
+	output.ServiceNamespace = HostingDeploymentAutoscalingServiceNamespace
+
+	return output, nil
+}
+
+// CreateDeleteScalingPolicyInput creates input
+func CreateDeleteScalingPolicyInput(spec hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJobSpec, resourceID string) applicationautoscaling.DeleteScalingPolicyInput {
+
+	if input, err := createDeleteScalingPolicyInput(spec, resourceID); err == nil {
+		return input
+	} else {
+		panic("Unable to create CreateDegisterScalableTargetInput " + err.Error())
+	}
+}
+
+// createDeleteScalingPolicyInput request input from a Kubernetes spec.
+func createDeleteScalingPolicyInput(spec hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJobSpec, resourceID string) (applicationautoscaling.DeleteScalingPolicyInput, error) {
+	var output applicationautoscaling.DeleteScalingPolicyInput
+
+	marshalledScalableDimension, err := json.Marshal(spec.ScalableDimension)
```

Comment:
This logic needs to be tested @mbaijal

Reply:
Marshal/Unmarshal throws scary errors that crash the controller image (sometimes indefinitely) if not handled properly if the specifications don't match.

Reply:
I guess this needs a bigger refactor to move away from using this design from the operators. any suggestions for now ?